packages/openneuro-server/src/graphql/__tests__/comment.spec.js

```diff
@@ -0,0 +1,52 @@
+import mockingoose from 'mockingoose'
+import { deleteComment, flatten } from '../resolvers/comment'
+import Comment from '../../models/comment'
+
+const incrementHex = (hex, step = 1) => {
+  hex = hex.toString()
+  const splitLen = 14
+  const start = hex.substr(0, splitLen)
+  const end = hex.substr(splitLen)
+  let dec = parseInt(end, 16)
+  dec++
+  return start + dec.toString(16)
+}
+
+describe('comment resolver helpers', () => {
+  describe('deleteComment', () => {
+    let aId
+    let commentA, commentB, commentC
+    beforeEach(async done => {
+      mockingoose.resetAll()
+      commentA = {
+        text: 'a',
+        createDate: new Date().toISOString(),
+        user: { __typename: 'User', _id: 'userid' },
+      }
+      mockingoose.Comment.toReturn(commentA, 'findOne')
+      aId = (await Comment.findById('just getting commentA here').exec())._id
+      mockingoose.Comment.toReturn([], 'find')
+      done()
+    })
+
+    it('returns an array of the deleted comment ids', async done => {
+      const deletedIds = (await deleteComment({}, { commentId: aId })).map(id =>
+        id.toString(),
+      )
+
+      // mockingoose ids are sequential, and the documents returned from each query get new ids for some reason
+      //   TODO: Replace mockingoose with better library
```

Comment:
Yeah, it doesn't 100% do what we want. I haven't looked recently to see if there's better ways to mock out MongoDB with Mongoose. I think mocking it is preferred to running a real MongoDB for tests but we could consider that as well.

Reply:
The odd thing is that we have [mongo-mock](https://www.npmjs.com/package/mongo-mock) in the test code already, but it's a bit sparse on features so mongoose was easier to implement in this case. [mongodb-memory-server](https://github.com/nodkz/mongodb-memory-server) looks promising, and it's already got good jest integration.